activation/poet.go

```diff
@@ -492,15 +491,14 @@ func (c *poetService) Submit(
 		if err != nil {
 			return nil, fmt.Errorf("recertifying: %w", err)
 		}
-		return c.client.Submit(submitCtx, deadline, prefix, challenge, signature, nodeID, *auth)
+		return c.client.Submit(ctx, deadline, prefix, challenge, signature, nodeID, *auth)
+	case errors.Is(err, context.DeadlineExceeded):
+		logger.Warn("failed to submit challenge: probably submit challenge timeout is too short", zap.Error(err))
 	}
 	return nil, fmt.Errorf("submitting challenge: %w", err)
 }
 
 func (c *poetService) Proof(ctx context.Context, roundID string) (*types.PoetProof, []types.Hash32, error) {
-	getProofsCtx, cancel := withConditionalTimeout(ctx, c.requestTimeout)
-	defer cancel()
-
```

Comment:
Fetching the proof still needs the timeout I believe 🙂. PoST proofing won't start until all proofs were fetched or timed out. After some time (18 mins is the default for mainnet I believe) the node should give up trying to fetch a proof from a PoET that doesn't respond and start PoST proving or it might run out of time within the cyclegap to publish an ATX and register for the next round.

Only registering shouldn't have a timeout (just the deadline of the PoET round start).